### PR TITLE
Add comments, get lint to be green, remove useless code

### PR DESCRIPTION
**Pull request by kenden**
(using this as a Golang training too :) )

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -17,36 +18,41 @@ import (
...
 	AWSSecretAccessKey string
 	AWSSessionToken    string
 	AWSAssumeRoleArn   string
 	AWSDefaultProfile  string
 }
 
+// NewConfig returns an empty Config
 func NewConfig() *Config {
 	return &Config{
-		AWSAssumeRoleArn:  "",
```
Comment (kenden) [inline main.go:36]: This should not be necessary, the default value is already ""

File: main.go [hunk 2]
```diff
@@ -156,6 +166,7 @@ func writeTempConfig(c *config.Config, config *Config, tmp string) error {
 	return c.WriteFile(tmp, 0644, "Updated by Boynux authenticator")
 }
 
+// swapFiles renames file orig into file temp
```
Comment (boynux) [inline main.go]: It's the other way around, write temp file into original ;)

File: main.go [hunk 3]
```diff
@@ -132,22 +140,24 @@ func getSessionToken(
...
 func writeTempConfig(c *config.Config, config *Config, tmp string) error {
 	c.AddSection("default")
-	c.AddOption("default", "aws_access_key_id", config.AWSAccessKeyId)
+	c.AddOption("default", "aws_access_key_id", config.AWSAccessKeyID)
 	c.AddOption("default", "aws_session_token", config.AWSSessionToken)
 	c.AddOption("default", "aws_secret_access_key", config.AWSSecretAccessKey)
 	c.AddOption("default", "role_arn", config.AWSAssumeRoleArn)
 	c.AddOption("default", "source_profile", config.AWSDefaultProfile)
 
+	// role_arn and source_profile only make sense if A
```
Comment (boynux) [inline main.go]: Seems incomplete line :?

**Comment by kenden** [comment]
@boynux fixed!

**Comment by boynux** [comment]
Fixes #4

**Comment by boynux** [comment]
Fixes #2

**Pull request merged**